lib/http_delete.js

```diff
@@ -0,0 +1,179 @@
+ * @return {Any} whatever callback returns
+ */
+function _handleErrors(opContext, errorAgent, errorObj,
+                       callback, currentError) {
+    const payload = {
+        topic: httpUtils.topics.delete,
+        key: opContext.fragments.objectKey,
+        messages: [JSON.stringify(errorObj)],
+    };
+
+    errorAgent.send([payload], err => {
+        if (err) {
+            opContext.log.error(
+                'Failure to persist orphaned fragments', err);
+            // Return persistence error
+            // TODO: not sure this a proper HTTP error....
+            // TODO: must be a better way than disabling it
+            /* eslint-disable no-param-reassign */
+            err.infos = { status: 500, method: 'DELETE' };
+            /* eslint-enable no-param-reassign */
+            return callback(err);
+        }
+
+        /* Error persistence OK, return already encountered
+         * error (might be null) */
+        return callback(currentError);
+    });
+}
+
+/**
+ * Delete a single fragment
+ *
+ * @param {http.HttpAgent} httpAgent - Agent to use
+ * @param {Object} errorAgent - Agent to persist errors
+ * @param {Object} opContext - Operation context
+ * @param {HyperdriveClient~deleteCallback} callback - Callback
+ * @param {Number} chunkId - Current chunk number
+ * @param {Number} fragmenId - Current fragment number
+ * @param {Number} requestTimeoutMs - Timeout of each sub-query
+ * @return {undefined}
+ */
+function fragmentDELETE(
+    { httpAgent, errorAgent, opContext, callback,
+      chunkId, fragmentId, requestTimeoutMs }) {
+    const isData = fragmentId < opContext.fragments.nDataParts;
+    const chunk = opContext.fragments.chunks[chunkId];
+    const fragment = isData ? chunk.data[fragmentId] :
+              chunk.coding[fragmentId - opContext.fragments.nDataParts];
+
+    const reqContext = { opContext, chunkId, fragmentId };
+    const { hostname, port, key } = fragment;
+    const requestOptions = httpUtils.getCommonStoreRequestOptions(
+        httpAgent, hostname, port, key);
+
+    requestOptions.method = 'DELETE';
+    requestOptions.headers = {
+        ['Content-Length']: 0,
+        ['Accept']: protocol.helpers.makeAccept(),
+    };
+
+    const request = httpUtils.newRequest(
+        requestOptions, opContext.log, reqContext, requestTimeoutMs,
+        /* callback */
+        reqCtx => {
+            const opCtx = reqCtx.opContext;
+            /* Wait for all to answer */
+            if (opCtx.nPending !== 0) {
+                return null;
+            }
+
+            let ret = null;
+            const { error, errorHandler } = decideDELETE(opCtx);
+            if (errorHandler.fragments.length === 0) {
+                ret = callback(error);
+            } else {
+                ret = _handleErrors(opContext, errorAgent,
+                                    errorHandler,
+                                    callback, error);
+            }
+            opCtx.log.end();
```

Comment:
It is a requestLogger. Why? Is it because end() just adds the mandatory fileds (like elapsed time) but odes not log anything? (I have this pattern everywhere ><)